beacon/beacon.go

```diff
@@ -1139,3 +1145,32 @@ func (pd *ProtocolDriver) gatherMetricsData() ([]*metrics.BeaconStats, *metrics.
 
 	return observed, calculated
 }
+
+// beaconClock provides methods to determine the intended send time of messages
+// spceific to the beacon protocol.
+type beaconClock struct {
```

Comment:
maybe rename this to something like messageTimer or something?

in this codebase, using the name `clock` implies it's ticking and driving something (as the layer is time-based).
but here you are using it to derive msg times.

Reply:
I'm not sure about using timer because go has a timer and that is an object which does drive something, but clock is also not great, let me think about it :thinking:

Reply:
Ok renamed to `beaconMessageTimes`

Reply:
perfect.